testng-core/src/main/java/org/testng/internal/MethodInheritance.java

```diff
@@ -135,14 +134,13 @@ public static void fixMethodInheritance(ITestNGMethod[] methods, boolean before)
                 ITestNGMethod m1 = l.get(i);
                 for (int j = i + 1; j < l.size(); j++) {
                   ITestNGMethod m2 = l.get(j);
-                  boolean groupMode = XmlTest.isGroupBasedExecution(m2.getXmlTest());
-                  if (groupMode) {
-                    // Do not resort to adding implicit depends-on if there are groups
-                    continue;
-                  }
                   if (!equalsEffectiveClass(m1, m2) && !dependencyExists(m1, m2, methods)) {
                     Utils.log("MethodInheritance", 4, m2 + " DEPENDS ON " + m1);
-                    m2.addMethodDependedUpon(MethodHelper.calculateMethodCanonicalName(m1));
+                    if (m1.getPriority() == 0
+                        && m2.getPriority() == 0
+                        && m1.getPriority() >= m2.getPriority()) {
```

Comment:
I dont see how m1.getPriority() >= m2.getPriority() can happen if it was already comfirmed that both ==0

Reply:
No, it can not happen. that was my first implementation, then I change to only check if both are 0. I don't want to risk modify this in case the users have configured priority.

Reply:
I guess the best would be to have a separate priority implementation for class hierarchy to avoid interfere with the user configuration.

Reply:
Well, the `>=` test will always be `true`, so it can be removed.

Reply:
Hi @cbeust 
Yes, an alternative solution can be found in #2668. When I now which solution is preferred I will cancel the others and rework the PR. Maybe it is to risky to reuse the priority and therefore I created a separate PR with a separate priority.